### DISCUSSION TITLE
Want to know more information on forget password and verify token route for unknown user.

### DISCUSSION THREAD

**Discussion by BhuwanPandey** [fastapi-users/fastapi-users#1216] [Q&A]
Hi there,
when i pass valid email address on auth/forget-password or auth/request-verify-token route 
we get token as argument on usermanager method . Is there any way to return response like shown on below .
It will be quite easy to send token on frontend if that might possible..

sync def on_after_forgot_password(
        self, user: User, token: str, request: Optional[Request] = None
    ):
        # print(f"User {user.id} has forgot their password. Reset token: {token}")
         return {"token":token}

    async def on_after_request_verify(
        self, user: User, token: str, request: Optional[Request] = None
    ):
      return {"token":token}
        

And also
Whether user provide valid or invalid email address, it always return null with status code of 202.
I want to return  message for invalid email. How do i do that?

**Comment by frankie567** [comment]
You *shouldn't* return the token in the API response. Doing this, you would totally defeat the purpose of reset password / email verification: anybody could enter an email, get a reset password token and use it to reset the password of any user. The strength of this technique is that we consider the email a safe place in control by the user.

As for getting an error response if the email does not exist, you have to reimplement the route yourself if you want to. That said, it's not recommended to do so as it would allow malicious users to enumerate existing accounts.
